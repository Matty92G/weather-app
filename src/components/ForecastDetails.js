import React from "react";

function ForecastDetails({ forecasts }) {
  return (
    <div className="forecast-details">
      <p>Date is {forecasts.date}</p>
      <p>description is {forecasts.description}</p>
      <p>icon is {forecasts.icon}</p>
      <p>temperatureMax is {forecasts.temperature.max}</p>
      <p>temperatureMin is {forecasts.temperature.min}</p>
      <p>windSpeed is {forecasts.wind.speed}</p>
      <p>windDirection is {forecasts.wind.direction}</p>
    </div>
  );
}

export default ForecastDetails;
