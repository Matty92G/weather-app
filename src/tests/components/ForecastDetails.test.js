import React from "react";
import { getByText, render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 11111111,
    description: "stubDescription",
    icon: "800",
    temperature: {
      min: 10,
      max: 99,
    },
    wind: {
      speed: 10,
      direction: "stubDirection",
    },
  };
  it("renders and matches snapshot", () => {
    const { asFragment } = render(
      <div className="forecast-details">
        <p>Date is {validProps.date}</p>
        <p>description is {validProps.description}</p>
        <p>icon is {validProps.icon}</p>
        <p>temperatureMax is {validProps.temperature.max}</p>
        <p>temperatureMin is {validProps.temperature.min}</p>
        <p>windSpeed is {validProps.wind.speed}</p>
        <p>windDirection is {validProps.wind.direction}</p>
      </div>
    );
    expect(asFragment).toMatchSnapshot();
  });
});
