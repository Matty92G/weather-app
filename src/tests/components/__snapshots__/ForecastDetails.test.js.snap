// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders and matches snapshot 1`] = `[Function]`;
